stonesoup/types/state.py

```diff
@@ -426,47 +427,85 @@ def __init__(self, *args, **kwargs):
             self.tag = str(uuid.uuid4())
 
 
-class ParticleState(Type):
+class ParticleState(State):
     """Particle State type
 
     This is a particle state object which describes the state as a
     distribution of particles"""
 
-    particles: Particles = Property(doc='All particles.')
+    weight: MutableSequence[Probability] = Property(default=None, doc='Weights of particles')
+    parent: 'ParticleState' = Property(default=None, doc='Parent particles')
+    particle_list: MutableSequence[Particle] = Property(default=None,
+                                                        doc='List of Particle objects')
     fixed_covar: CovarianceMatrix = Property(default=None,
                                              doc='Fixed covariance value. Default `None`, where'
                                                  'weighted sample covariance is then used.')
-    timestamp: datetime.datetime = Property(default=None,
-                                            doc="Timestamp of the state. Default None.")
 
-    def __init__(self, particles, *args, **kwargs):
-        if particles is not None and not isinstance(particles, Particles):
-            particles = Particles(particle_list=particles)
-        super().__init__(particles, *args, **kwargs)
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if (self.particle_list is not None) and \
+                (self.state_vector is not None or self.weight is not None):
+            raise ValueError("Use either a list of Particle objects or StateVectors and weights,"
+                             " but not both.")
+
+        if self.particle_list and isinstance(self.particle_list, list):
+            self.state_vector = \
+                StateVectors([particle.state_vector for particle in self.particle_list])
+            self.weight = \
+                np.array([Probability(particle.weight) for particle in self.particle_list])
+            parent_list = [particle.parent for particle in self.particle_list]
+
+            if parent_list.count(None) == 0:
+                self.parent = ParticleState(particle_list=parent_list)
+            elif 0 < parent_list.count(None) < len(parent_list):
+                raise ValueError("Either all particles should have"
+                                 " parents or none of them should.")
+
+        if self.parent:
+            self.parent.parent = None
```

Comment:
This was to avoid using significant memory, as all particles would be linked. By setting parent's parent to `None`, it should allow old particles to be removed by garbage collection. Whilst we don't use parent attribute yet, in discussions there are approaches where referring to direct parent can be useful.

Reply:
@sdhiscocks Perhaps adding a comment to indicate why this was done would be helpful. The question would likely come up again when someone else looks at the code.